server/src/models/userModel.js

```diff
@@ -10,7 +10,22 @@ const userSchema = new Schema(
     },
     email: {
       type: String,
-      required: true,
+      lowercase: true,
+      match: [/(.+)@(.+){2,}\.(.+){2,}/, "Please enter a valid email"],
+      validate: {
+        validator: async function (email) {
+          const user = await this.constructor.findOne({ email });
+          if (user) {
+            if (this.id === user.id) {
+              return true;
+            }
+            return false;
+          }
+          return true;
+        },
+        message: () => "Email address already registered.",
+      },
```

Comment:
any specific reason we haven't used a unique attribute here? I feel its better to add unique attribute here and if validation errors are all handled at one single place rather than doing it here in the model itself.

Reply:
Using `unique` option in mongoose would basically check the email field index for uniqueness. But the shortcoming of this approach is that the index must be created on the database for this to work. If index is not created, unique would not work. This is pointed out [here](https://mongoosejs.com/docs/validation.html#the-unique-option-is-not-a-validator).

Reply:
should we put this check inside the static logging method at the bottom of the file?

Reply:
Hii @lugenx ,why are we adding validation to the userModel route we can simply check this in the userController route. 

![image](https://user-images.githubusercontent.com/106306508/211472984-f3cc6148-3587-4631-9932-ac241750b0a6.png)

I am new to the open source and just learning how to contribute, I hope it helps.

Reply:
yeah, this is another way of doing that, and i think it would work just fine, but i still would prefer to keep user validations in the user model.

Reply:
> inside the static method

We can do that. Let me know if you'd like me to make any change other than this and removing the regex.